hal/ia32/cpu.c

```diff
@@ -175,21 +155,21 @@ int hal_cpuReschedule(spinlock_t *spinlock, spinlock_ctx_t *scp)
 		"pushl $0;"
 		"pushl %%eax;"
 		"pushl $0;"
-		"movl %4, %%eax;"
+		"movl %1, %%eax;"
 		"call *%%eax;"
 		"cli;"
 		"addl $12, %%esp;"
 		"popl %%esp;"
 
 		"call _interrupts_multilockClear;"
- 		"pushl %%esp;"
+		"pushl %%esp;"
 
 		"jmp interrupts_popContext;"
 
-		"3:;"
+		"1:;"
 		"movl %%eax, %0"
 	: "=g" (err)
```

Comment:
```suggestion
	: "=a" (err)
```
You avoid contruction type:
```
mov %eax, %ecx
mov %ecx, %eax
```
If you accept this suggestion remove `eax` from `clobbers`.

Reply:
I agree, I also removed last line `movl %%eax, %0` as it's no longer needed.